lib/rubocop/cop/rails/where_missing.rb

```diff
@@ -50,7 +52,15 @@ def on_send(node)
         private
 
         def root_receiver(node)
-          node&.parent&.send_type? ? root_receiver(node.parent) : node
+          if node&.parent&.send_type?
+            if node.parent.method?(:or) || node.parent.method?(:and)
+              node
+            else
+              root_receiver(node.parent)
+            end
+          else
+            node
+          end
```

Comment:
Can this code be refactored as below?

```suggestion
          parent = node.parent
          if !parent&.send_type? || parent.method?(:or) || parent.method?(:and)
            node
          else
            root_receiver(parent)
          end
```

Reply:
I updated it. Thank you!